### ISSUE TITLE
Reading response content in request involving redirects throws COM exception

### ISSUE THREAD

**Issue by GunArm** [MicrosoftEdge/WebView2Feedback#1839]
**Description**
When trying to get the literal response text with `CoreWebView2.WebResourceResponseReceived`, if the request involves an http redirect (for example 302 or 307 to redirect from http to https), the e.Response.GetContentAsync() method throws an exception:
```
System.Runtime.InteropServices.COMException: The pipe is being closed. (0x800700E8)
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHR(Int32 errorCode)
   at Microsoft.Web.WebView2.Core.CoreWebView2WebResourceResponseView.GetContentAsync()
```

It is not only that it throws this exception on the first request which gives the 300 to redirect (as if one did not account for the redirect and only tried to read the redirected response content) -- if a 300 is involved, both the http 302 and the https 200 each throw this exception when trying to await GetContentAsync() on their respective responses.

**Version**
SDK: 1.0.992.28
Runtime: 94.0.992.38
Framework: Winforms
OS: Win10

**Repro Steps**
Small reproducing sample: https://pastebin.com/J4faFM6Y
Try with both http and https urls for www.google.com
Https url gets content successfully.
Http (307 -> https) throws an exception trying to read content from both requests.

[AB#36791649](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/36791649)

**Comment by gudieaofei** [comment]
@champnic tow thousand years later

**Comment by monica-ch** [comment]
@GunArm Issue has been fixed in SDK version: [1.0.1150.38](https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.1150.38) and requires WebView2 Runtime 99.0.1150.38 or higher.

However, we throw an exception if the response body is empty for 307 redirect and other https request with 200 response code should give the content.
